backends/dpdk/dpdkArch.cpp

```diff
@@ -2773,4 +2773,135 @@ const IR::Node* ElimHeaderCopy::postorder(IR::Member* m) {
+        return false;
+    else
+        return true;
+}
+
+const IR::Node* DpdkAddPseudoHeaderDecl::preorder(IR::P4Control* c) {
+    auto ctrl = getOriginal<IR::P4Control>();
+    for (auto p : ctrl->type->applyParams->parameters) {
+        if (auto tn = p->type->to<IR::Type_Name>()) {
+            auto decl = typeMap->getTypeType(tn, true);
+            if (isHeadersStruct(decl->to<IR::Type_Struct>())) {
+                headersDeclName = p->name.name;
+            }
+        }
+    }
+    return c;
+}
+
+std::pair<IR::AssignmentStatement*, IR::Member*>
+MoveNonHeaderFieldsToPseudoHeader::addAssignmentStmt(const IR::NamedExpression* ne) {
+    auto name = refMap->newName("pseudo");
+    pseudoFieldNameType.push_back(std::pair(name, typeMap->getType(ne->expression, true)));
+    auto mem0 =
+        new IR::Member(new IR::PathExpression(IR::ID(DpdkAddPseudoHeaderDecl::headersDeclName)),
+                       IR::ID(headerInstanceName));
+    auto mem1 = new IR::Member(mem0, IR::ID(name));
+    return {new IR::AssignmentStatement(mem1, ne->expression), mem1};
+}
+
+const IR::Node* MoveNonHeaderFieldsToPseudoHeader::postorder(IR::MethodCallStatement* statement) {
+    auto mce = getOriginal<IR::MethodCallStatement>()->methodCall;
+    bool added_copy = false;
+    IR::Vector<IR::Argument> arguments;
+    IR::StructExpression* struct_exp = nullptr;
+    auto result = new IR::IndexedVector<IR::StatOrDecl>();
+    if (auto* m = mce->method->to<IR::Member>()) {
+        if (auto* type = typeMap->getType(m->expr)->to<IR::Type_Extern>()) {
+            if (type->name == "InternetChecksum") {
+                if (m->member == "add" || m->member == "subtract") {
+                    auto components = new IR::IndexedVector<IR::NamedExpression>();
+                    for (auto arg : *mce->arguments) {
+                        const IR::StructExpression* tmp =
+                            arg->expression->to<IR::StructExpression>();
+                        if (tmp) {
+                            for (auto c : arg->expression->to<IR::StructExpression>()->components) {
+                                if (auto m = c->expression->to<IR::Member>()) {
+                                    // is it coming from metadata, move it to the pseudo header
+                                    if (!typeMap->getType(m->expr, true)->is<IR::Type_Header>()) {
+                                        added_copy = true;
+                                        auto stm = addAssignmentStmt(c);
+                                        result->push_back(stm.first);
+                                        components->push_back(new IR::NamedExpression(
+                                            c->srcInfo, c->name, stm.second));
+                                    } else
+                                        components->push_back(c);
+                                } else {
+                                    // is it constant? move it also to the pseduo header
```

Comment:
You are not explicitly checking for constant here. So this comment doesn't  make much sense.
I think the code is fine because we should be handling even local variable if those are not yet moved to metadata. Please update the comment